src/compiler/checker.ts

```diff
@@ -337,6 +337,16 @@ module ts {
                             break loop;
                         }
                         break;
+                    case SyntaxKind.ComputedPropertyName:
+                        var grandparent = location.parent.parent;
+                        if (grandparent.kind === SyntaxKind.ClassDeclaration || grandparent.kind === SyntaxKind.InterfaceDeclaration) {
+                            // A reference to this grandparent's type parameters would be an error
+                            if (result = getSymbol(getSymbolOfNode(grandparent).members, name, meaning & SymbolFlags.Type)) {
+                                error(errorLocation, Diagnostics.A_computed_property_name_cannot_reference_a_type_parameter_from_its_containing_type);

```

Comment:
Yes, a good example is tests\cases\conformance\es6\computedProperties\computedPropertyNames32.ts:
```ts
function foo<T>() { return '' }
class C<T> {
    bar() {
        return 0;
    }
    [foo<T>()]() { }
}
```
I will add this snippet in a comment in the code as well.